spring-batch-integration/src/main/java/org/springframework/batch/integration/partition/RemotePartitioningWorkerStepBuilder.java

```diff
@@ -0,0 +1,249 @@
+ *     on the input channel</li>
+ *     <li>invokes the {@link StepExecutionRequestHandler} to execute the worker
+ *     step for each incoming request. The worker step is located using the provided
+ *     {@link StepLocator}. If no {@link StepLocator} is provided, a {@link BeanFactoryStepLocator}
+ *     configured with the current {@link BeanFactory} will be used
+ *     <li>replies to the master on the output channel (when the master step is
+ *     configured to aggregate replies from workers). If no output channel
+ *     is provided, a {@link NullChannel} will be used</li>
+ * </ul>
+ *
+ * @since 4.1
+ * @author Mahmoud Ben Hassine
+ */
+public class RemotePartitioningWorkerStepBuilder extends StepBuilder {
+
+	private static final String SERVICE_ACTIVATOR_METHOD_NAME = "handle";
+
+	private MessageChannel inputChannel;
+	private MessageChannel outputChannel;
+	private JobExplorer jobExplorer;
+	private StepLocator stepLocator;
+	private BeanFactory beanFactory;
+
+	/**
+	 * Initialize a step builder for a step with the given name.
+	 * @param name the name of the step
+	 */
+	public RemotePartitioningWorkerStepBuilder(String name) {
+		super(name);
+	}
+
+	/**
+	 * Set the input channel on which step execution requests sent by the master
+	 * are received.
+	 * @param inputChannel the input channel
+	 * @return this builder instance for fluent chaining
+	 */
+	public RemotePartitioningWorkerStepBuilder inputChannel(MessageChannel inputChannel) {
+		Assert.notNull(inputChannel, "inputChannel must not be null");
+		this.inputChannel = inputChannel;
+		return this;
+	}
+
+	/**
+	 * Set the output channel on which replies will be sent to the master step.
+	 * @param outputChannel the input channel
+	 * @return this builder instance for fluent chaining
+	 */
+	public RemotePartitioningWorkerStepBuilder outputChannel(MessageChannel outputChannel) {
+		Assert.notNull(outputChannel, "outputChannel must not be null");
+		this.outputChannel = outputChannel;
+		return this;
+	}
+
+	/**
+	 * Set the job explorer.
+	 * @param jobExplorer the job explorer to use
+	 * @return this builder instance for fluent chaining
+	 */
+	public RemotePartitioningWorkerStepBuilder jobExplorer(JobExplorer jobExplorer) {
+		Assert.notNull(jobExplorer, "jobExplorer must not be null");
+		this.jobExplorer = jobExplorer;
+		return this;
+	}
+
+	/**
+	 * Set the step locator used to locate the worker step to execute.
+	 * @param stepLocator the step locator to use
+	 * @return this builder instance for fluent chaining
+	 */
+	public RemotePartitioningWorkerStepBuilder stepLocator(StepLocator stepLocator) {
+		Assert.notNull(stepLocator, "stepLocator must not be null");
+		this.stepLocator = stepLocator;
+		return this;
+	}
+
+	/**
+	 * Set the bean factory.
+	 * @param beanFactory the bean factory
+	 * @return this builder instance for fluent chaining
+	 */
+	public RemotePartitioningWorkerStepBuilder beanFactory(BeanFactory beanFactory) {
```

Comment:
Why are the required fields (jobExplorer, etc) not configured via the constructor?

Reply:
To be consistent with other builders.